File: LeetCode/349.intersection-of-two-arrays.java
import java.util.Set;
import java.util.HashSet;
import java.util.Collections;
/*
 * @lc app=leetcode id=349 lang=java
 *
 * [349] Intersection of Two Arrays
 */

// @lc code=start
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {

        int n1 = nums1.length;
        int n2 = nums2.length;

        Set<Integer> st1 = new HashSet<>();
        Set<Integer> st2 = new HashSet<>();

        for (int i : nums1) {
            st1.add(i);
        }

        for (int i : nums2) {
            st2.add(i);
        }

        st1.retainAll(st2);

        int[] ans = new int[st1.size()];
        int i = 0;
        for (int it : st1) {
            ans[i++] = it;
        }

        return ans;

    }
}
// @lc code=end

File: LeetCode/349.intersection-of-two-arrays.py
#
# @lc app=leetcode id=349 lang=python3
#
# [349] Intersection of Two Arrays
#

# @lc code=start
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:

        # nums1 = set(nums1)
        # nums2 = set(nums2)

        # return list(nums1 & nums2)


        #  res =[]

        # for i in nums1:
        #     if i not in res and i in nums2:
        #         res.append(i)

        # return res


      
       

        res =[]

        nums1.sort()
        nums2.sort()

        i=j=0

        while( i<len(nums1) and j<len(nums2)):
            if nums1[i] == nums2[j]:
                if nums1[i] not in res:
                    res.append(nums1[i])
                i+=1
                j+=1

            elif nums1[i]<nums2[j]:
                 i+=1

            else:
                j+=1    


        return res

        








        
# @lc code=end

